logic/auth.js

```diff
@@ -189,6 +189,43 @@ async function getInfo() {
     }
 };
 
+async function getSettings() {
+    try {
+        const defaultSettings = await diskLogic.readDefaultSettingsFile();
```

Comment:
I don't really understand your point. If we add a new settings in the future, we will just need to add its default value to the default file. The manager will take care to load default settings and add them to the current user settings when receiving a request. What do you want to handle in particular in the manager?

Having an external file sounds better for me and makes future contributions to the dashboard more easy. Someone implementing a new feature won't have to explore the manager code to find where to add its setting´s default value, and just add it to the file, which path is explicitly specified in the docs.

Reply:
Sorry I don't think I explained that very clearly and also this was incorrect:

>AND read the current default file for existing users and update it inline.

The other points still stand though.

Basically my thinking is that the default settings are static data specific to manager so it belongs in the manager repo.

The only files that belong in the Umbrel repo are dynamic files that need to persist between container lifetimes.

The `templates/` dir is not really designed to be read from directly. It's used as an input for the configure script that will generate the resulting files in container specific volumes so they live between container lifetimes. The default settings are only relevant to the manager codebase so it doesn't make sense for them to live in a different code base and then be passed back in to manager at runtime.

Also passing things around via environment variables and volumes has already got a bit out of hand. I think that needs cleaning up and reducing. So if we can implement something in a simpler way and without adding more variables/volumes that need to be passed around that seems like the better solution to me.

That's why I think the default values belong in manager.

I guess it could be a JSON file in the manager repo. I don't have a huge issue against this but personally I'd always prefer less code complexity over more. An inline object seems much simpler than an external file and extra functions to read it and potentially handle errors.

Does that make sense?

Reply:
> The only files that belong in the Umbrel repo are dynamic files that need to persist between container lifetimes.
>
>The templates/ dir is not really designed to be read from directly. It's used as an input for the configure script that will generate the resulting files in container specific volumes so they live between container lifetimes. The default settings are only relevant to the manager codebase so it doesn't make sense for them to live in a different code base and then be passed back in to manager at runtime.
>
>Also passing things around via environment variables and volumes has already got a bit out of hand. I think that needs cleaning up and reducing. So if we can implement something in a simpler way and without adding more variables/volumes that need to be passed around that seems like the better solution to me.

That totally makes sense, thanks for the explanation!

> I don't have a huge issue against this but personally I'd always prefer less code complexity over more. An inline object seems much simpler than an external file and extra functions to read it and potentially handle errors.

I understand what you want to say, particularly about the code complexity of a file reading over an inline object, but I can't figure that the latter is better than the former in terms of code clarity. See:
>Having an external file sounds better for me and makes future contributions to the dashboard more easy. Someone implementing a new feature won't have to explore the manager code to find where to add its setting´s default value, and just add it to the file, which path is explicitly specified in the docs.

Maybe we can compromise and import the default settings json at the start of `auth.js` and not read it each time (as it should not change)?
```
import defaultSettings from 'blabla/default-settings.json'
```
If I do that, under which directory it's better to save this file?

Reply:
Hmmnn, there don't seem to be any existing directories that really make sense. Maybe `resources/` if we had to pick one, creating a new dir just for this file seems excessive.

I guess it could go in the project root, although I generally think it's cleaner to keep as much stuff out of root as possible.

Which do you prefer?

Reply:
Let's go for `resources`!

Do my other changes (latest commits) + the use of an `import` statement suit you?

Reply:
I think in the manager readme or even not at all. Anyone looking to edit it will find the require in auth.js.

Reply:
OKay 👌